packages/react-dom/src/server/ReactPartialRenderer.js

```diff
@@ -764,8 +824,62 @@ class ReactDOMServerRenderer {
           );
         // eslint-disable-next-line-no-fallthrough
         default:
-          return this.renderDOM(nextElement, context, parentNamespace);
+          break;
       }
+      if (typeof elementType === 'object' && elementType !== null) {
+        switch (elementType.$$typeof) {
+          case REACT_PROVIDER_TYPE: {
+            const provider: ReactProvider<any> = (nextChild: any);
+            const nextProps = provider.props;
+            const nextChildren = toArray(nextProps.children);
+            const frame: Frame = {
+              type: provider,
+              domNamespace: parentNamespace,
+              children: nextChildren,
+              childIndex: 0,
+              context: context,
+              footer: '',
+            };
+            if (__DEV__) {
+              ((frame: any): FrameDev).debugElementStack = [];
```

Comment:
Can someone explain this syntax please?

Reply:
This is a way to do a relatively unsafe typecast with flow https://flow.org/

```jsx
(variableReference: type);
```

We can’t do the following to cast `frame` to a `FrameDev` because flow already understands it to be of type `Frame`:

```jsx
(frame: FrameDev).debugElementStack = [];
```

By first saying `(frame: any)` we effectively erase the knowledge that it is a `Frame` to be able to say specifically that it is of type `FrameDev`.

Definitely grain of salt the specifics on that regarding how flow internally represents this, but that should give you a high level view of what’s going on.

Reply:
Thanks!